rero_ils/modules/patrons/permissions.py

```diff
@@ -128,16 +129,68 @@ def delete(cls, user, record):
         return cls.create(user, record)
 
 
-def get_allowed_roles_management():
+def get_allowed_roles_management(patron_pid=None):
     """Get the roles that current logged user could manage.
 
+    :param patron_pid: the patron pid to check
     :return An array of allowed role management.
     """
     allowed_roles = []
-    if current_patron and current_patron.is_librarian:
-        if current_patron.is_librarian:
-            allowed_roles.append(Patron.ROLE_PATRON)
-            allowed_roles.append(Patron.ROLE_LIBRARIAN)
-        if current_patron.is_system_librarian:
-            allowed_roles.append(Patron.ROLE_SYSTEM_LIBRARIAN)
+    patron = Patron.get_record_by_pid(patron_pid) if patron_pid else None
+    restrictions = current_app.config.get('ROLES_MANAGEMENT_PERMISSIONS', {})
+    for role in Patron.ALL_ROLES:
+        # by default, all operations are allowed
+        data = {
+            'name': role,
+            'permissions': {
+                'add': {'can': ALLOW},
+                'delete': {'can': ALLOW}
+            }
+        }
+        # check about role management restrictions to know if one action should
+        # be denied + reasons
+        for name, conditions in restrictions.get(role, {}).items():
+            reasons = [condition.message for condition in conditions
+                       if not condition.can(patron)]
+            # If reasons array isn't empty, the action is denied
+            if reasons:
+                data['permissions'][name]['can'] = DENY
+                data['permissions'][name]['reasons'] = reasons
+        allowed_roles.append(data)
     return allowed_roles
+
+
+# =============================================================================
+# CONDITIONS
+# =============================================================================
+
+class StaffMemberCondition(AbstractCondition):
+    """Condition class to check if the current patron is a staff member.
+
+    To be considerate as a staff member the patron should have one of the
+    role defined into Patron.STAFF_MEMBERS_ROLE.
```

Comment:
```suggestion
    To be considered as a staff member the patron must have at least one of the
    role defined into Patron.STAFF_MEMBERS_ROLE.
```
I'm not sure about the `at least`. Your call.